api/types/src/transaction.rs

```diff
@@ -557,11 +570,18 @@ pub struct VersionedEvent {
 impl From<(&EventWithVersion, serde_json::Value)> for VersionedEvent {
     fn from((event, data): (&EventWithVersion, serde_json::Value)) -> Self {
         match &event.event {
-            ContractEvent::V0(v0) => Self {
+            ContractEvent::V1(v1) => Self {
+                version: event.transaction_version.into(),
+                guid: (*v1.key()).into(),
+                sequence_number: v1.sequence_number().into(),
+                typ: v1.type_tag().clone().into(),
+                data,
+            },
```

Comment:
not really. Without that naming change, I still to add those code everywhere because `Deref` is not working anymore since we have a second discriminant.